File: src/main/java/com/leon/bugreport/listeners/UpdateChecker.java
package com.leon.bugreport.listeners;

import com.leon.bugreport.logging.ErrorMessages;
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.scheduler.BukkitRunnable;

import java.io.IOException;
import java.io.InputStream;
import java.net.URL;
import java.util.Scanner;
import java.util.function.Consumer;

import static com.leon.bugreport.API.ErrorClass.logErrorMessage;

public class UpdateChecker {
	private final JavaPlugin plugin;
	private final int resourceId;

	public UpdateChecker(JavaPlugin plugin, int resourceId) {
		this.plugin = plugin;
		this.resourceId = resourceId;
	}

	public void getVersion(final Consumer<String> consumer) {
		new BukkitRunnable() {
			@Override
			public void run() {
				try (InputStream is = new URL("https://api.spigotmc.org/legacy/update.php?resource=" + resourceId).openStream()) {
					try (Scanner scanner = new Scanner(is)) {
						if (scanner.hasNext()) {
							consumer.accept(scanner.next());
						}
					}
				} catch (IOException e) {
					String errorMessage = ErrorMessages.getErrorMessage(33);

					plugin.getLogger().warning(errorMessage);
					logErrorMessage(errorMessage);
				}
			}
		}.runTaskAsynchronously(plugin);
	}
}

File: src/main/java/com/bugreportmc/bugreport/listeners/UpdateChecker.kt
package com.bugreportmc.bugreport.listeners

import org.bukkit.plugin.Plugin
import org.bukkit.scheduler.BukkitRunnable
import java.io.IOException
import java.net.URL
import java.util.*
import java.util.function.Consumer

class UpdateChecker(private val plugin: Plugin, private val resourceId: Int) {
	fun getVersion(consumer: Consumer<String>) {
		object : BukkitRunnable() {
			override fun run() {
				try {
					URL("https://api.spigotmc.org/legacy/update.php?resource=$resourceId").openStream().use { `is` ->
						Scanner(`is`).use { scanner ->
							if (scanner.hasNext()) {
								consumer.accept(scanner.next())
							}
						}
					}
				} catch (e: IOException) {
					plugin.logger.warning("Unable to check for updates: " + e.message)
				}
			}
		}.runTaskAsynchronously(plugin)
	}
}
